lib/plugins/features/movies/controller.js

```diff
@@ -27,3 +30,19 @@ exports.list = (query) => {
   })
   .fetchAll();
 };
+
+exports.addLocation = (payload) => {
+  return Bluebird.all([
+    new Location({ id: payload.location_id }).fetch(),
+    new Movie({ id: payload.movie_id }).fetch()
+  ])
+  .then((result) => {
+    const movie = result[1];
+    return movie.locations().attach(payload.location_id);
```

Comment:
Instead of referring to the relationship between movies and locations through the function `.location`, you should use `.related('locations')`. This is the preferred way to access the relations.